src/policy/ephemeral_policy.h

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_POLICY_EPHEMERAL_POLICY_H
+#define BITCOIN_POLICY_EPHEMERAL_POLICY_H
+
+#include <policy/packages.h>
+#include <policy/policy.h>
+#include <primitives/transaction.h>
+#include <txmempool.h>
+
+/** Does context-less checks about a single transaction.
+ * If it has relay dust, it returns false if any are true:
+ *  - tx has non-0 fee
+    - tx has more than one dust output
+ * and sets relevant invalid state.
+ * Otherwise it returns true.
+ */
+bool CheckValidEphemeralTx(const CTransaction& tx, CFeeRate dust_relay_fee, CAmount txfee, TxValidationState& state);
+
+/** Checks that all dust in package ends up spent by direct children. Assumes package is well-formed and sorted. */
+std::optional<uint256> CheckEphemeralSpends(const Package& package, CFeeRate dust_relay_rate);
```

Comment:
In 872c06dacde6c45c2add69c8c15c94571b08119d "policy: Allow dust in transactions, spent in-mempool"

Could you document what the return value is?

Reply:
documented